agent/agent.go

```diff
@@ -93,6 +95,9 @@ func (a *Agent) loop(ctx context.Context) error {
 		select {
 		case <-ctx.Done():
 			grip.Info("agent loop canceled")
+			if a.statusServer != nil {
+				a.statusServer.Shutdown(ctx)
+			}
```

Comment:
I think that you wait on <-ctx.Done() and then call this in the same thread that starts the server to avoid having to entangle the status server into the agent struct.

Reply:
Done (switched to using graceful)